bob-common/src/metrics/collector/accumulator.rs

```diff
@@ -28,13 +27,10 @@ impl MetricsAccumulator {
     // themselves, not just references
     #[allow(clippy::needless_pass_by_value)]
     pub(crate) async fn run(mut self) {
-        let mut interval = interval(self.interval);
         loop {
-            interval.tick().await;
             let mut metrics_received = 0;
             let timestamp = get_current_unix_timestamp();
-
-            while let Ok(om) = timeout(METRICS_RECV_TIMEOUT, self.rx.recv()).await {
+            while let Ok(om) = timeout(self.interval, self.rx.recv()).await {
```

Comment:
This looks like an incorrect change. As I understand it: the initial logic was to update `snapshot` once in an `interval`. With the current change updates become non-deterministic. To make a fix correct, you need to add below (in a line `if metrics_received == MAX_METRICS_PER_PERIOD {`) additional check that time since the last update becomes greater than `interval`. Also for the `timeout` method normally you should reduce interval parameter in every iteration, but this gives an overhead, so it is better to keep previous `METRICS_RECV_TIMEOUT` as a timeout value

Reply:
I think those changes were made before MAX_METRICS_PER_SECOND was introduced. I've reverted this changes since they have nothing to do with the purpose of this branch